### DISCUSSION TITLE
Questions about emulation

### DISCUSSION THREAD

**Discussion by volm92** [InterDigitalInc/AdvantEDGE#285] [Q&A] [answered]
Good morning! 

I want to ask you if there is any difference, besides the name, in the PoA types during emulation (WiFi, 4G, 5G, etc).
What is emulated different based on the type of PoA I choose? Does anything change in the data flow? Is the control plane emulate in somehow?
Also how is the RTT calculated? Or is it measured run time?

Thankyou! (yes I read wiki but sometimes I've some doubt, I'm sorry if I ask too much :D)

**Comment by dilallkx** [comment, marked as answer]
Hi @volm92,

Thank you for the questions!

There is no difference in emulation for the PoA types. Network characteristics are applied the same way for all of them and are used to apply end-to-end traffic shaping rules.

PoA Types are used in 2 places:
- For automated PoA selection by the terminal
  - Terminal configuration allows you to define the Supported Wireless Types (order by priority)
  - In a scenario with multiple PoA types the terminal will attach to the highest priority, in-range PoA
  - NOTE: automated PoA selection can be enabled by selecting Mobility in the Event Automation pane
    - https://github.com/InterDigitalInc/AdvantEDGE/wiki/usage-execution-view#automation-pane
- By the MEC Service API implementations (e.g. RNIS & WAIS) 
  - Some APIs include information about the PoA types in API responses
  - PoA types are also used to calculate emulated signal strengths to return in API responses
  - API documentation can be found here: https://github.com/InterDigitalInc/AdvantEDGE/wiki/overview-api

For the RTT (latencies in network metrics graphs), these are measured values using ping round-trip time.

Hope this helps!
Regards!

**Reply by volm92** [reply]
Thanks a lot for the answer!

>In a scenario with multiple PoA types the terminal will attach to the highest priority, in-range PoA

What happens if I do not activate automatic PoA selection, but the terminal is under the cover of multiple PoAs? To which one does it attach and according to what criterion?

>PoA types are also used to calculate emulated signal strengths to return in API responses

In what way? I haven't been able to get a value that means signal strength yet. Which PoA supports this information?
Is it possible to acquire signal strength with RNIS API?

> For the RTT (latencies in network metrics graphs), these are measured values using ping round-trip time.

Ok! So they are measured. Could this be the reason why even if I put jitter = 0 I still get a fluctuation in latency? 
In case I set the jitter to zero and have fluctuations in latency, could it be due to the hardware?

Thank you a lot!
Marco

**Reply by dilallkx** [reply]
Hi @volm92,

Without automatic PoA selection the terminal remains attached to the parent PoA it is configured with in the scenario. You can send manual MOBILITY events to move the terminal from one PoA to another.

For the PoA signal strengths, you can get these from the following API calls:
- WAI API:
  - /queries/sta/sta_information --> returned StaInfo structure contains RSSI.
  - NOTE: your terminal must be in range & attached to a WiFi POA
-  RNI API:
  - You must subscribe for MeasRep (4G) or NrMeasRep (5G) notifications using the /subscriptions endpoint
  - The signal strength information is passed via the RSRP & RSRQ values returned in the notifications
  - NOTE: your terminal must be in range of a Cellular PoA

Note that the signal strength values are calculated using the terminal location within the PoA range; it is therefore necessary to have properly configured PoA location & radius, as well as terminal location.

For the latency measurements, yes, your deployment hardware could impact the results. It is also important to note that to avoid jitter you must make sure that the full end-to-end path (including all nodes from source node to destination node) is configured with 0 jitter.

Hope this helps!
Regards!
